src/main/java/org/skriptlang/skript/lang/converter/ChainedConverter.java

```diff
@@ -40,27 +40,45 @@
  */
 final class ChainedConverter<F, M, T> implements Converter<F, T> {
 
-	private final Converter<F, M> first;
-	private final Converter<M, T> second;
+	private final ConverterInfo<F, M> first;
+	private final ConverterInfo<M, T> second;
 
-	ChainedConverter(Converter<F, M> first, Converter<M, T> second) {
+	ChainedConverter(ConverterInfo<F, M> first, ConverterInfo<M, T> second) {
 		this.first = first;
 		this.second = second;
 	}
 
 	@Override
 	@Nullable
 	public T convert(F from) {
-		M middle = first.convert(from);
+		M middle = first.getConverter().convert(from);
 		if (middle == null) {
 			return null;
 		}
-		return second.convert(middle);
+		return second.getConverter().convert(middle);
 	}
 
 	@Override
 	public String toString() {
-		return "ChainedConverter{first=" + first + ",second=" + second + "}";
+		StringBuilder builder = new StringBuilder();
+
+		builder.append("ChainedConverter{(");
+		if (first.getConverter() instanceof ChainedConverter) {
+			builder.append(first.getConverter());
+		} else {
+			builder.append(first.getFrom()).append(" -> ").append(first.getTo());
+		}
+
+		builder.append(") -> (");
+
+		if (second.getConverter() instanceof ChainedConverter) {
+			builder.append(second.getConverter());
+		} else {
+			builder.append(second.getFrom()).append(" -> ").append(second.getTo());
```

Comment:
Regular Converter doesn't have a toString override, so this is not a useful toString.

Reply:
`toString` is only used on ChainedConverters. Regular converters will just show `<from> -> <to>` so it should be alright?